apps/alpha-test-app/src/components/layout-demo/LayoutSettings.vue

```diff
@@ -0,0 +1,165 @@
+);
+
+const emit = defineEmits<{
+  "update:modelValue": [value: SettingsSections];
+}>();
+
+const noneOption = { id: "none", label: "None" };
+
+const contentOptions: SelectionOption<undefined>[] = [
+  { id: "longPageContent", label: "Scrollable Page Content" },
+  { id: "showFlyout", label: "Flyout" },
+  { id: "forceTooltip", label: "Tooltip" },
+  { id: "showStickyContent", label: "Sticky Content" },
+  { id: "showToast", label: "Toast" },
+];
+const overlayOptions: SelectionOption<undefined>[] = [
+  noneOption,
+  { id: "showPopover", label: "Popover/Modal" },
+  { id: "showMobileFlyIn", label: "Mobile Fly-in" },
+  { id: "showPageLoader", label: "Page loader" },
+  { id: "topBarFlyout", label: "Top bar flyout" },
+];
+const sidebarOptions: SelectionOption<undefined>[] = [
+  noneOption,
+  { id: "showSideBar", label: "Fixed Sidebar" },
+  { id: "showSideBarCollapse", label: "Collapsible Sidebar" },
+  { id: "showTempOverlay", label: "Overlay Sidebar with backdrop" },
+  { id: "showTempOverlayTransparent", label: "Overlay Sidebar" },
+];
+const detailFooter = { id: "detailFooter", label: "Detail Footer" };
+const fullFooter = { id: "fullFooter", label: "Full Footer" };
+const baseFooterOptions = [noneOption, fullFooter];
+const sidebarFooterOptions = [noneOption, detailFooter, fullFooter];
+const footerOptions = ref<SelectionOption<undefined>[]>(sidebarFooterOptions);
+
+const activeContentSetting = computed({
+  get: () => props.modelValue.content || {},
+  set: (value) => emit("update:modelValue", { ...props.modelValue, content: value }),
+});
+const activeSidebarSetting = computed({
+  get: () => props.modelValue.sideBar || {},
+  set: (value) => emit("update:modelValue", { ...props.modelValue, sideBar: value }),
+});
+const activeFooterSetting = computed({
+  get: () => props.modelValue.footer || {},
+  set: (value) => emit("update:modelValue", { ...props.modelValue, footer: value }),
+});
+const activeOverlaySetting = computed({
+  get: () => props.modelValue.overlay || {},
+  set: (value) => emit("update:modelValue", { ...props.modelValue, overlay: value }),
+});
+
+const showSections = computed(() => {
+  const sectionsToShow = props.show.reduce<Partial<Record<SettingsSection, boolean>>>(
+    (visibleSections, section) => {
+      visibleSections[section] = true;
+      return visibleSections;
+    },
+    {},
+  );
+  if (
+    activeSidebarSetting.value.showTempOverlay ||
+    activeSidebarSetting.value.showTempOverlayTransparent
+  ) {
+    // block overlays when the temporary overlay is open
+    sectionsToShow.overlay = false;
+  }
+  return sectionsToShow;
+});
+
+/** Adust footer configs depending on the availability of a sidebar */
+watch(
+  activeSidebarSetting,
+  (sidebarSetting) => {
+    if (!sidebarSetting.showSideBar && !sidebarSetting.showSideBarCollapse) {
+      footerOptions.value = baseFooterOptions;
+    } else {
+      footerOptions.value = sidebarFooterOptions;
+    }
+  },
+  { immediate: true },
+);
```

Comment:
The footerOptions could be a computed right instead of ref + watcher. It is not used anywhere else